community/phantom-embedded-js/README.md

```diff
@@ -0,0 +1,199 @@
+# Phantom Embedded JS
+
+A minimal starter template for integrating **Phantom's embedded user wallets** on Solana using vanilla JavaScript/TypeScript. This template demonstrates how to authenticate users with Phantom Connect and display their Solana account information.
+
+## Features
+
+- **Phantom Connect Authentication** - OAuth-based user authentication with Google and Apple
+- **Embedded User Wallet** - No browser extension required
+- **Real-time Balance Display** - View SOL balance with auto-refresh every 30 seconds
+- **Direct Blockchain Queries** - Uses @solana/web3.js to query Solana RPC directly
+- **Copy Address** - One-click address copying
+- **Theme Switching** - Toggle between light and dark modes (light mode default)
+- **Fully Responsive** - Works seamlessly on mobile and desktop
+- **Fast Development** - Built with Vite for instant hot reload
+- **Persistent Preferences** - Theme preference saved in localStorage
+
+## Getting Started
+
+### Prerequisites
+
+- Node.js 18+ or Bun
+- pnpm, npm, or yarn
+- A Phantom App ID from [Phantom Developer Portal](https://phantom.com/portal)
+
+### Installation
+
+```bash
+# Create from template
+npx create-solana-dapp my-app --template phantom-embedded-js
+
+# Navigate to project
+cd my-app
+
+# Install dependencies
+pnpm install
+
+# Configure environment variables
+cp .env.example .env
+# Edit .env and add your Phantom App ID
+
+# Run development server
+pnpm dev
+```
+
+The app will open at [http://localhost:5173](http://localhost:5173)
+
+### Environment Setup
+
+Edit the `.env` file and add your Phantom App ID:
+
+```env
+# Your App ID from Phantom Portal (REQUIRED)
+VITE_PHANTOM_APP_ID=your-app-id-here
+
+# Phantom Connect Auth URL (optional - defaults to https://connect.phantom.app)
+VITE_PHANTOM_AUTH_URL=https://connect.phantom.app
+
+# Authentication Redirect URL (must be whitelisted in Phantom Portal)
+VITE_REDIRECT_URL=http://localhost:5173/
+
+# Solana RPC Endpoint (optional - defaults to devnet)
+VITE_SOLANA_RPC_URL=https://api.devnet.solana.com
+```
+
+## Configuration
+
+### Get Your Phantom App ID
+
+1. Visit [phantom.com/portal](https://phantom.com/portal) to create an app and get your App ID
+2. Whitelist your redirect URL: `http://localhost:5173/` (for development)
```

Comment:
Best practices should include a /auth/callback url for redirect URL. 

To better handle the states of failure/success and loading.

It should also mention to whitelist your domain URL